spec/datadog/tracing/contrib/action_pack/action_dispatch/journey/router_spec.rb

```diff
@@ -0,0 +1,233 @@
+        'BooksController',
+        Class.new(ActionController::Base) do
+          def show
+            head :ok
+          end
+        end
+      )
+    end
+
+    let(:status_controller) do
+      stub_const(
+        'StatusesController',
+        Class.new(ActionController::Base) do
+          def show
+            head :ok
+          end
+        end
+      )
+    end
+
+    let(:engine) do
+      stub_const('AuthEngine', Module.new)
+
+      stub_const(
+        'AuthEngine::TokensController',
+        Class.new(ActionController::Base) do
+          def create
+            head :ok
+          end
+        end
+      )
+
+      stub_const(
+        'AuthEngine::Engine',
+        Class.new(::Rails::Engine) do
+          isolate_namespace AuthEngine
+        end
+      )
+    end
+
+    context 'with default configuration' do
+      before do
+        Datadog.configure do |c|
+          c.tracing.instrument :rack
+          c.tracing.instrument :action_pack
+        end
+
+        clear_traces!
+      end
+
+      it 'sets http.route when requesting a known route' do
+        get '/api/users/1'
+
+        request_span = spans.first
+
+        expect(last_response).to be_ok
+        expect(request_span.name).to eq('rack.request')
+        expect(request_span.tags.fetch('http.route')).to eq('/api/users/:id')
+        expect(request_span.tags).not_to have_key('http.route.path')
+      end
+
+      it 'sets http.route correctly for ambiguous route with constraints' do
+        get '/items/1'
+
+        request_span = spans.first
+
+        expect(last_response).to be_ok
+        expect(request_span.name).to eq('rack.request')
+        expect(request_span.tags.fetch('http.route')).to eq('/items/:id')
+        expect(request_span.tags).not_to have_key('http.route.path')
+      end
+
+      it 'sets http.route correctly for ambiguous route with constraints, case two' do
+        get '/items/something'
+
+        request_span = spans.first
+
+        expect(last_response).to be_ok
+        expect(request_span.name).to eq('rack.request')
+        expect(request_span.tags.fetch('http.route')).to eq('/items/:slug')
+        expect(request_span.tags).not_to have_key('http.route.path')
+      end
+
+      it 'sets http.route correctly for routes with globbing' do
+        get 'books/some/section/title'
+
+        request_span = spans.first
+
+        expect(last_response).to be_ok
+        expect(request_span.name).to eq('rack.request')
+        expect(request_span.tags.fetch('http.route')).to eq('/books/*section/:title')
+        expect(request_span.tags).not_to have_key('http.route.path')
+      end
+
+      it 'sets http.route and http.route.path for rails engine routes' do
+        get '/api/auth/sign-in'
+
+        request_span = spans.first
+
+        expect(last_response).to be_ok
+        expect(request_span.name).to eq('rack.request')
+        expect(request_span.tags.fetch('http.route')).to eq('/api/auth/sign-in(/:expires_in)')
```

Comment:
This looks to test an optional route parameter which is a great test to have, I would suggest explicitly declaring at least one of the tests as testing the optional parameter so that wouldn't be accidentally removed in the future.

Reply:
fixed [3207b16](https://github.com/DataDog/dd-trace-rb/pull/3849/commits/3207b16ea39d9a5b7af0d7a1ae392058d2c39656)